lambdas/utils/aws/s3.py

```diff
@@ -24,24 +24,26 @@ def parse_s3_uri(s3_uri: str) -> tuple[str, str]:
             raise ValueError(f"Invalid S3 URI scheme: {s3_uri}")
 
         bucket = parsed.netloc
-        key = parsed.path.lstrip("/")
+        key = parsed.path.removeprefix("/")
         return bucket, key
 
     @classmethod
     def folder_exists(cls, s3_uri: str) -> bool:
         """Ensure that an S3 folder exists.
 
-        This is performed by looking for an object that exists under this prefix,
+        This is performed by looking for an object that begins with this prefix,
         effectively demonstrating this is an S3 "folder" (though really just a prefix).
+        Theoretically, it's possible this could be a zero-byte object "folder" and NOT
+        contain any child objects, but this would still satisfy this method of detecting
+        a "folder-like" object at this URI.
```

Comment:
Thanks for adding more detail! I agree with the description. I think my initial confusion was "What if a user passes in an S3 URI to a file and not a folder?", however, I do think the use-case will become clearer as the `S3Client` is put to use!